web/src/api/types.ts

```diff
@@ -0,0 +1,12 @@
+import type { Configuration } from './open-api';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ApiFp = (configuration: Configuration) => Record<any, (...args: any) => any>;
+
+export type OmitLast<T extends readonly unknown[]> = T extends readonly [...infer U, any?]
```

Comment:
Can you help explain this for educational purposes? Is there any resource I can read on about these derive?

Reply:
I've used [Variadic Tuple Types](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#variadic-tuple-types) for `OmitLast` to work. This type is used in `ApiParams` to pass in the parameters of the original function and remove the last parameter, in this case `options`.